source/templates.tex

```diff
@@ -262,19 +263,95 @@
   \opt{nested-name-specifier} concept-name \terminal{<} \opt{template-argument-list} \terminal{>}
 \end{bnf}
 
+\begin{bnf}
+\nontermdef{type-tt-parameter}\br
+  template-head type-parameter-key \opt{\terminal{...}} \opt{identifier}\br
+  template-head type-parameter-key \opt{identifier} \opt{type-tt-parameter-default}
+\end{bnf}
+
+\begin{bnf}
+\nontermdef{type-tt-parameter-default}\br
+  \terminal{=} \opt{nested-name-specifier} template-name\br
+  \terminal{=} nested-name-specifier \terminal{template} template-name
+\end{bnf}
+
+\begin{bnf}
+\nontermdef{variable-tt-parameter}\br
+  template-head \terminal{auto} \opt{\terminal{...}} \opt{identifier}\br
+  template-head \terminal{auto} \opt{identifier} \terminal{=} \opt{nested-name-specifier} template-name
+\end{bnf}
+
+\begin{bnf}
+\nontermdef{concept-tt-parameter}\br
+  template \terminal{<} template-parameter-list \terminal{>} \terminal{concept} \opt{\terminal{...}} \opt{identifier}\br
+  template \terminal{<} template-parameter-list \terminal{>} \terminal{concept} \opt{identifier} \terminal{=} \opt{nested-name-specifier} template-name
+\end{bnf}
+
 \indextext{component name}%
 The component names of a \grammarterm{type-constraint} are
 its \grammarterm{concept-name} and
 those of its \grammarterm{nested-name-specifier} (if any).
 \begin{note}
 The \tcode{>} token following the
 \grammarterm{template-parameter-list} of a
-\grammarterm{type-parameter}
+\grammarterm{type-tt-parameter},
+\grammarterm{variable-tt-parameter}, or
+\grammarterm{concept-tt-parameter}
 can be the product of replacing a
 \tcode{>>} token by two consecutive \tcode{>}
 tokens\iref{temp.names}.
 \end{note}
 
+\pnum
+%FIXME: "is" or "shall be"? i.e., what if it's not?
```

Comment:
is. It's a close set. I do agree we never seem to define _template parameter_, this might be the right place to do it.

Reply:
Could you send a separate pull request to create such a definition?